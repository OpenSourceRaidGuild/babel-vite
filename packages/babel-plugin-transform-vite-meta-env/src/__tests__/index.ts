import pluginTester from 'babel-plugin-tester'
import plugin from '..'

pluginTester({
  plugin,
  pluginName: 'vite-meta-env',
  snapshot: true,
  tests: {
    'replace MODE': 'const x = import.meta.env.MODE',
    'replace BASE_URL': 'const x = import.meta.env.BASE_URL',
    'replace NODE_ENV': 'const x = import.meta.env.NODE_ENV',
    'replace DEV': 'const x = import.meta.env.DEV',
    'replace PROD': 'const x = import.meta.env.PROD',
    'replace VITE_* variables': 'const x = import.meta.env.VITE_VAR',
    'replace string access': 'const x = import.meta.env["VITE_VAR"]',
    'replace key access': 'const key = "VITE_VAR"; const x = import.meta.env[key]',
    'replace env object': 'const env = import.meta.env',
    'not replaceable': 'const x = import.meta.env.OTHER',
    'not import.meta.env': 'const x = import.meta.other',
    'not import.meta': 'const x = process.env.MODE',
    'not import.meta lookup': 'const x = import.meta()'
  }
})

pluginTester({
  plugin,
  pluginName: 'vite-meta-env',
  pluginOptions: {
    envPrefix: 'TEST_PREFIX_'
  },
  snapshot: true,
  tests: {
    'custom prefix replace MODE': 'const x = import.meta.env.MODE',
    'custom prefix replace BASE_URL': 'const x = import.meta.env.BASE_URL',
    'custom prefix replace NODE_ENV': 'const x = import.meta.env.NODE_ENV',
    'custom prefix replace DEV': 'const x = import.meta.env.DEV',
    'custom prefix replace PROD': 'const x = import.meta.env.PROD',
    'custom prefix replace TEST_PREFIX_* variables': 'const x = import.meta.env.TEST_PREFIX_VAR',
    'custom prefix replace string access': 'const x = import.meta.env["TEST_PREFIX_VAR"]',
    'custom prefix replace key access':
      'const key = "TEST_PREFIX_VAR"; const x = import.meta.env[key]',
    'custom prefix replace env object': 'const env = import.meta.env',
    'custom prefix not replaceable': 'const x = import.meta.env.OTHER',
    'custom prefix not replaceable VITE_': 'const x = import.meta.env.VITE_VAR',
    'custom prefix not import.meta.env': 'const x = import.meta.other',
    'custom prefix not import.meta': 'const x = process.env.MODE',
    'custom prefix not import.meta lookup': 'const x = import.meta()'
  }
})

pluginTester({
  plugin,
  pluginName: 'vite-meta-env',
  pluginOptions: {
    envPrefix: ['TEST_PREFIX_A_', 'TEST_PREFIX_B_']
  },
  snapshot: true,
  tests: {
    'multiple custom prefix replace MODE': 'const x = import.meta.env.MODE',
    'multiple custom prefix replace BASE_URL': 'const x = import.meta.env.BASE_URL',
    'multiple custom prefix replace NODE_ENV': 'const x = import.meta.env.NODE_ENV',
    'multiple custom prefix replace DEV': 'const x = import.meta.env.DEV',
    'multiple custom prefix replace PROD': 'const x = import.meta.env.PROD',
    'multiple custom prefix replace TEST_PREFIX_A_* variables':
      'const x = import.meta.env.TEST_PREFIX_A_VAR',
    'multiple custom prefix replace TEST_PREFIX_B_* variables':
      'const x = import.meta.env.TEST_PREFIX_B_VAR',
    'multiple custom prefix replace string access': 'const x = import.meta.env["TEST_PREFIX_A"]',
    'multiple custom prefix replace key access A':
      'const key = "TEST_PREFIX_A_VAR"; const x = import.meta.env[key]',
    'multiple custom prefix replace key access B':
      'const key = "TEST_PREFIX_B_VAR"; const x = import.meta.env[key]',
    'multiple custom prefix replace env object': 'const env = import.meta.env',
    'multiple custom prefix not replaceable': 'const x = import.meta.env.OTHER',
    'multiple custom prefix not replaceable VITE_': 'const x = import.meta.env.VITE_VAR',
    'multiple custom prefix not import.meta.env': 'const x = import.meta.other',
    'multiple custom prefix not import.meta': 'const x = process.env.MODE',
    'multiple custom prefix not import.meta lookup': 'const x = import.meta()'
  }
})
