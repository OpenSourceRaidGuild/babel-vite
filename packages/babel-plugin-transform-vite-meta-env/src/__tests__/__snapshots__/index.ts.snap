// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vite-meta-env custom prefix not import.meta lookup: custom prefix not import.meta lookup 1`] = `

const x = import.meta()

      ↓ ↓ ↓ ↓ ↓ ↓

const x = import.meta()


`;

exports[`vite-meta-env custom prefix not import.meta.env: custom prefix not import.meta.env 1`] = `

const x = import.meta.other

      ↓ ↓ ↓ ↓ ↓ ↓

const x = import.meta.other


`;

exports[`vite-meta-env custom prefix not import.meta: custom prefix not import.meta 1`] = `

const x = process.env.MODE

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.MODE


`;

exports[`vite-meta-env custom prefix not replaceable VITE_: custom prefix not replaceable VITE_ 1`] = `

const x = import.meta.env.VITE_VAR

      ↓ ↓ ↓ ↓ ↓ ↓

const x = {
  ...Object.fromEntries(Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_'))),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}.VITE_VAR


`;

exports[`vite-meta-env custom prefix not replaceable: custom prefix not replaceable 1`] = `

const x = import.meta.env.OTHER

      ↓ ↓ ↓ ↓ ↓ ↓

const x = {
  ...Object.fromEntries(Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_'))),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}.OTHER


`;

exports[`vite-meta-env custom prefix replace BASE_URL: custom prefix replace BASE_URL 1`] = `

const x = import.meta.env.BASE_URL

      ↓ ↓ ↓ ↓ ↓ ↓

const x = '/'


`;

exports[`vite-meta-env custom prefix replace DEV: custom prefix replace DEV 1`] = `

const x = import.meta.env.DEV

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV !== 'production'


`;

exports[`vite-meta-env custom prefix replace MODE: custom prefix replace MODE 1`] = `

const x = import.meta.env.MODE

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV || 'test'


`;

exports[`vite-meta-env custom prefix replace NODE_ENV: custom prefix replace NODE_ENV 1`] = `

const x = import.meta.env.NODE_ENV

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV || 'test'


`;

exports[`vite-meta-env custom prefix replace PROD: custom prefix replace PROD 1`] = `

const x = import.meta.env.PROD

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV === 'production'


`;

exports[`vite-meta-env custom prefix replace TEST_PREFIX_* variables: custom prefix replace TEST_PREFIX_* variables 1`] = `

const x = import.meta.env.TEST_PREFIX_VAR

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.TEST_PREFIX_VAR


`;

exports[`vite-meta-env custom prefix replace env object: custom prefix replace env object 1`] = `

const env = import.meta.env

      ↓ ↓ ↓ ↓ ↓ ↓

const env = {
  ...Object.fromEntries(Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_'))),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}


`;

exports[`vite-meta-env custom prefix replace key access: custom prefix replace key access 1`] = `

const key = "TEST_PREFIX_VAR"; const x = import.meta.env[key]

      ↓ ↓ ↓ ↓ ↓ ↓

const key = 'TEST_PREFIX_VAR'
const x = {
  ...Object.fromEntries(Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_'))),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}[key]


`;

exports[`vite-meta-env custom prefix replace string access: custom prefix replace string access 1`] = `

const x = import.meta.env["TEST_PREFIX_VAR"]

      ↓ ↓ ↓ ↓ ↓ ↓

const x = {
  ...Object.fromEntries(Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_'))),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}['TEST_PREFIX_VAR']


`;

exports[`vite-meta-env multiple custom prefix not import.meta lookup: multiple custom prefix not import.meta lookup 1`] = `

const x = import.meta()

      ↓ ↓ ↓ ↓ ↓ ↓

const x = import.meta()


`;

exports[`vite-meta-env multiple custom prefix not import.meta.env: multiple custom prefix not import.meta.env 1`] = `

const x = import.meta.other

      ↓ ↓ ↓ ↓ ↓ ↓

const x = import.meta.other


`;

exports[`vite-meta-env multiple custom prefix not import.meta: multiple custom prefix not import.meta 1`] = `

const x = process.env.MODE

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.MODE


`;

exports[`vite-meta-env multiple custom prefix not replaceable VITE_: multiple custom prefix not replaceable VITE_ 1`] = `

const x = import.meta.env.VITE_VAR

      ↓ ↓ ↓ ↓ ↓ ↓

const x = {
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_A_'))
  ),
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_B_'))
  ),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}.VITE_VAR


`;

exports[`vite-meta-env multiple custom prefix not replaceable: multiple custom prefix not replaceable 1`] = `

const x = import.meta.env.OTHER

      ↓ ↓ ↓ ↓ ↓ ↓

const x = {
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_A_'))
  ),
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_B_'))
  ),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}.OTHER


`;

exports[`vite-meta-env multiple custom prefix replace BASE_URL: multiple custom prefix replace BASE_URL 1`] = `

const x = import.meta.env.BASE_URL

      ↓ ↓ ↓ ↓ ↓ ↓

const x = '/'


`;

exports[`vite-meta-env multiple custom prefix replace DEV: multiple custom prefix replace DEV 1`] = `

const x = import.meta.env.DEV

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV !== 'production'


`;

exports[`vite-meta-env multiple custom prefix replace MODE: multiple custom prefix replace MODE 1`] = `

const x = import.meta.env.MODE

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV || 'test'


`;

exports[`vite-meta-env multiple custom prefix replace NODE_ENV: multiple custom prefix replace NODE_ENV 1`] = `

const x = import.meta.env.NODE_ENV

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV || 'test'


`;

exports[`vite-meta-env multiple custom prefix replace PROD: multiple custom prefix replace PROD 1`] = `

const x = import.meta.env.PROD

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV === 'production'


`;

exports[`vite-meta-env multiple custom prefix replace TEST_PREFIX_A_* variables: multiple custom prefix replace TEST_PREFIX_A_* variables 1`] = `

const x = import.meta.env.TEST_PREFIX_A_VAR

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.TEST_PREFIX_A_VAR


`;

exports[`vite-meta-env multiple custom prefix replace TEST_PREFIX_B_* variables: multiple custom prefix replace TEST_PREFIX_B_* variables 1`] = `

const x = import.meta.env.TEST_PREFIX_B_VAR

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.TEST_PREFIX_B_VAR


`;

exports[`vite-meta-env multiple custom prefix replace env object: multiple custom prefix replace env object 1`] = `

const env = import.meta.env

      ↓ ↓ ↓ ↓ ↓ ↓

const env = {
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_A_'))
  ),
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_B_'))
  ),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}


`;

exports[`vite-meta-env multiple custom prefix replace key access A: multiple custom prefix replace key access A 1`] = `

const key = "TEST_PREFIX_A_VAR"; const x = import.meta.env[key]

      ↓ ↓ ↓ ↓ ↓ ↓

const key = 'TEST_PREFIX_A_VAR'
const x = {
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_A_'))
  ),
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_B_'))
  ),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}[key]


`;

exports[`vite-meta-env multiple custom prefix replace key access B: multiple custom prefix replace key access B 1`] = `

const key = "TEST_PREFIX_B_VAR"; const x = import.meta.env[key]

      ↓ ↓ ↓ ↓ ↓ ↓

const key = 'TEST_PREFIX_B_VAR'
const x = {
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_A_'))
  ),
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_B_'))
  ),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}[key]


`;

exports[`vite-meta-env multiple custom prefix replace string access: multiple custom prefix replace string access 1`] = `

const x = import.meta.env["TEST_PREFIX_A"]

      ↓ ↓ ↓ ↓ ↓ ↓

const x = {
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_A_'))
  ),
  ...Object.fromEntries(
    Object.entries(process.env).filter(([k]) => k.startsWith('TEST_PREFIX_B_'))
  ),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}['TEST_PREFIX_A']


`;

exports[`vite-meta-env not import.meta lookup: not import.meta lookup 1`] = `

const x = import.meta()

      ↓ ↓ ↓ ↓ ↓ ↓

const x = import.meta()


`;

exports[`vite-meta-env not import.meta.env: not import.meta.env 1`] = `

const x = import.meta.other

      ↓ ↓ ↓ ↓ ↓ ↓

const x = import.meta.other


`;

exports[`vite-meta-env not import.meta: not import.meta 1`] = `

const x = process.env.MODE

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.MODE


`;

exports[`vite-meta-env not replaceable: not replaceable 1`] = `

const x = import.meta.env.OTHER

      ↓ ↓ ↓ ↓ ↓ ↓

const x = {
  ...Object.fromEntries(Object.entries(process.env).filter(([k]) => k.startsWith('VITE_'))),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}.OTHER


`;

exports[`vite-meta-env replace BASE_URL: replace BASE_URL 1`] = `

const x = import.meta.env.BASE_URL

      ↓ ↓ ↓ ↓ ↓ ↓

const x = '/'


`;

exports[`vite-meta-env replace DEV: replace DEV 1`] = `

const x = import.meta.env.DEV

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV !== 'production'


`;

exports[`vite-meta-env replace MODE: replace MODE 1`] = `

const x = import.meta.env.MODE

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV || 'test'


`;

exports[`vite-meta-env replace NODE_ENV: replace NODE_ENV 1`] = `

const x = import.meta.env.NODE_ENV

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV || 'test'


`;

exports[`vite-meta-env replace PROD: replace PROD 1`] = `

const x = import.meta.env.PROD

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.NODE_ENV === 'production'


`;

exports[`vite-meta-env replace VITE_* variables: replace VITE_* variables 1`] = `

const x = import.meta.env.VITE_VAR

      ↓ ↓ ↓ ↓ ↓ ↓

const x = process.env.VITE_VAR


`;

exports[`vite-meta-env replace env object: replace env object 1`] = `

const env = import.meta.env

      ↓ ↓ ↓ ↓ ↓ ↓

const env = {
  ...Object.fromEntries(Object.entries(process.env).filter(([k]) => k.startsWith('VITE_'))),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}


`;

exports[`vite-meta-env replace key access: replace key access 1`] = `

const key = "VITE_VAR"; const x = import.meta.env[key]

      ↓ ↓ ↓ ↓ ↓ ↓

const key = 'VITE_VAR'
const x = {
  ...Object.fromEntries(Object.entries(process.env).filter(([k]) => k.startsWith('VITE_'))),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}[key]


`;

exports[`vite-meta-env replace string access: replace string access 1`] = `

const x = import.meta.env["VITE_VAR"]

      ↓ ↓ ↓ ↓ ↓ ↓

const x = {
  ...Object.fromEntries(Object.entries(process.env).filter(([k]) => k.startsWith('VITE_'))),
  NODE_ENV: process.env.NODE_ENV || 'test',
  MODE: process.env.NODE_ENV || 'test',
  BASE_URL: '/',
  DEV: process.env.NODE_ENV !== 'production',
  PROD: process.env.NODE_ENV === 'production'
}['VITE_VAR']


`;
